src/storage-folder.coffee

```diff
@@ -0,0 +1,27 @@
+path = require "path"
+fs = require "fs-plus"
+
+module.exports =
+class StorageFolder
+  constructor: (containingPath) ->
+    @path = path.join(containingPath, "storage")
+
+  store: (name, object) ->
+    fs.writeFileSync(@pathForKey(name), JSON.stringify(object), 'utf8')
```

Comment:
> I extracted this from Atom::saveSync()

Oh okay.

> What would the try/catch be for in this case?

I would think if `fs.writeFileSync` fails then atom wouldn't quit possibly since an error is thrown from the `beforeunload` handler that this runs in.

And since `store` is also called in the browser process, I wasn't sure how an error thrown during `before-quit` handler would behave.